core/modules/layout/layout.admin.inc

```diff
@@ -1129,6 +1130,133 @@ function layout_block_configure_redirect($layout, $block_uuid) {
+            $region_label = t(' in ') . check_plain($region_name) . t(' region');
+            $block_index = '';
+            if ($multiple_blocks) {
+              $block_index = t(' block ') . ($index + 1);
+            }
+            $options[$block->uuid] = $current_layout->title . $path . $region_label . $block_index;
+          }
+        }
+      }
+
+      $form['remove_blocks'] = array(
+        '#type' => 'checkboxes',
+        '#title' => t('Remove block %block_title from the following layouts (paths)', array('%block_title' => $block_title)),
+        '#options' => $options,
+        '#default_value' => array($block_uuid),
+      );
+    }
+  }
+
+  if (!isset($form['remove_blocks'])) {
+    $form['message'] = array(
+      '#markup' => t('Remove this block?'),
+    );
+    $form_state['layout'] = $layout;
+    $form_state['block_uuid'] = $block_uuid;
+  }
+
+  $form['actions'] = array('#type' => 'actions');
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Remove'),
+  );
+  $form['actions']['cancel'] = array(
+    '#type' => 'submit',
+    '#value' => t('Cancel'),
+    '#attributes' => array(
+      'class' => array('button-secondary',),
+    ),
+    '#submit' => array(
+      'layout_remove_block_cancel_action'
+    ),
+  );
+
+  return $form;
+}
+
+/**
+ * Submit handler for layout_block_contextual_remove_form() cancel button.
+ */
+function layout_remove_block_cancel_action(&$form, &$form_state) {
+  // Just call this to close the dialogue.
+}
+
+/**
+ * Submit handler for layout_block_contextual_remove_form() submit button.
+ */
+function layout_block_contextual_remove_form_submit($form, &$form_state) {
+  $block_title = $form_state['block_title'];
+  $removed = array();
+
+  if (isset($form_state['values']['remove_blocks'])) {
+    $block_uuids_to_remove = array_filter($form_state['values']['remove_blocks']);
+    if ($block_uuids_to_remove) {
+      $layouts = $form_state['layouts'];
+      foreach($layouts as $layout) {
+        foreach($block_uuids_to_remove as $remove_block) {
+          if (isset($layout->content[$remove_block])) {
+            $layout->removeBlock($remove_block);
+            $layout->save();
+
+            $removed[] = $layout->title;
+          }
+        }
+      }
+    }
+  }
+  else {
+    $layout = $form_state['layout'];
+    $layout->removeBlock($form_state['block_uuid']);
+    $layout->save();
+
+    $removed[] = $layout->title;
+  }
+
+  $message = '';
+  if (empty($removed)) {
+    $message = t('No blocks were removed');
+  }
+  elseif (count($removed) === 1) {
+    $message = t('Block %block_title was removed from layout %layout.', array('%block_title' => $block_title, '%layout' => reset($removed)));
+  }
+  else {
+    $message = t('Block %block_title was removed from the following layouts:!layouts', array('%block_title' => $block_title, '!layouts' => theme('item_list', array('items' => $removed))));
```

Comment:
I'd suggest just having `t()` around `'Block %block_title was removed from the following layouts:` and just append the `theme('item_list`)`.

It also looks like `$layout->title` is unsanitized within the item list. Using `array_map('check_plain', $removed)` should make all items safe.